### ISSUE TITLE
Safe Browsing Errors

### ISSUE THREAD

**Issue by mheubusch** [mozilla-mobile/fenix#2397]
As a user I want to know when I am about to access a site with malware or other damaging files or content so that  I  can stay safe online. 

Why/User Benefit/User Problem

Make sure that users understand why a website could not be loaded
    Present them with an error screen and a way to back out/ continue browsing to another website

What / Requirements

This is all about displaying an error page when a page did not load because of an error condition managed via the Safe Browsing partnership.

We have an Error Pages component that covers the following:

```
ERROR_SAFEBROWSING_MALWARE_URI
ERROR_SAFEBROWSING_UNWANTED_URI
ERROR_SAFEBROWSING_HARMFUL_URI
ERROR_SAFEBROWSING_PHISHING_URI
```

Acceptance Criteria (how do I know when I’m done?)
I can see content that informs me about the error

**Comment by sblatz** [comment]
Was blocked by https://github.com/mozilla-mobile/android-components/issues/2725

**Comment by sblatz** [comment]
Should be fixed by https://github.com/mozilla-mobile/fenix/pull/2686

**Comment by abodea** [comment]
Hello, note that I found 3 web pages where this issue still occurs on the latest **`build 1.0.1921(5/23)`** with Google Pixel 3(Android P) after I verified the following:
### Safe Browsing Testing [Links](https://testsafebrowsing.appspot.com/) scenarios
* [A/W/M/L/C] Should show a phishing warning: [link](https://testsafebrowsing.appspot.com/s/phishing.html)  - the error is displayed **`Deceptive site issue`**
* [A/W/M/L/C/D] Should show a malware warning: [link](https://testsafebrowsing.appspot.com/s/malware.html) - the error is displayed **`Harmful site issue`**
* [A/W/M/L/C/D] Should show a malware warning due to a bad subresource: [link](https://testsafebrowsing.appspot.com/s/malware_in_iframe.html) - there is no error displayed only a white page with **`Malware in iframe page example`** text. [logcat.txt](https://github.com/mozilla-mobile/fenix/files/3210953/logcat.txt) for step 3.
* [W/M/L/C] Should show a unwanted software warning: [link
](https://testsafebrowsing.appspot.com/s/unwanted.html) - the error is displayed **`Unwanted site issue`** 
* [W/M/L/C/D] Should show a malware warning due to bad assets: [ [small img](https://testsafebrowsing.appspot.com/s/image_small.html), [medium img](https://testsafebrowsing.appspot.com/s/image_medium.html), [large img](https://testsafebrowsing.appspot.com/s/image_large.html), [css](https://testsafebrowsing.appspot.com/s/bad_css.html), [js](https://testsafebrowsing.appspot.com/s/bad_javascript.html)] - there is no error displayed for the mentioned links, only a white screen. [logcat.txt](https://github.com/mozilla-mobile/fenix/files/3210987/logcat.txt) this is the logcat for step 4.
* [A/W/M/L/C] Should show a billing warning: [link](https://testsafebrowsing.appspot.com/s/trick_to_bill.html) - no error is displayed, only a white screen. [logcatt.txt](https://github.com/mozilla-mobile/fenix/files/3210991/logcatt.txt) this is the logcat for step 6.

**Comment by Amejia481** [comment]
Hi @abodea,

🤔  I'm not sure if we are handling  these cases, I tested on Fennec and **there is no error displayed** those, I will investigate further.
* [W/M/L/C/D] Should show a malware warning due to bad assets: [ [small img](https://testsafebrowsing.appspot.com/s/image_small.html), [medium img](https://testsafebrowsing.appspot.com/s/image_medium.html), [large img](https://testsafebrowsing.appspot.com/s/image_large.html), [css](https://testsafebrowsing.appspot.com/s/bad_css.html), [js](https://testsafebrowsing.appspot.com/s/bad_javascript.html)] - there is no error displayed for the mentioned links, only a white screen.

* [A/W/M/L/C/D] Should show a malware warning due to a bad subresource: [link](https://testsafebrowsing.appspot.com/s/malware_in_iframe.html) - there is no error displayed only a white page with **`Malware in iframe page example`.

* [A/W/M/L/C] Should show a billing warning: [link](https://testsafebrowsing.appspot.com/s/trick_to_bill.html) - no error is displayed, only a white screen.

**Comment by Amejia481** [comment]
Yeah, none of these categories are implemented for safe browsing, neither of them actually work on Fennec. But two on them work on Chrome:  **Warning due to bad assets and billing warning**.   If they are required they need work from GeckoView team. @vesta0 Should I open bug for this?

**Comment by vesta0** [comment]
@Amejia481 thanks for investigating! Since we have Fennec parity, we can close this issue. Would you please open a new issue for the new categories we don't yet support so we can prioritize for post-mvp, and also open a GeckoView bug on bugzilla so @cpeterso can track?

**Comment by Amejia481** [comment]
Sure!

- [A-C issue](https://github.com/mozilla-mobile/android-components/issues/3108)
- [GeckoView issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1554226)
